code/datums/datumvars.dm

```diff
@@ -362,55 +364,55 @@ function loadPage(list) {
 			(<a href='?_src_=vars;datumsave=\ref[DA];varnamesave=[name]'>S</a>) "}
 
 	if (isnull(value))
-		html += "[name] = <span class='value'>null</span>"
+		html += "[prefix]<span class='value'>null</span>"
 
 	else if (istext(value))
-		html += "[name] = <span class='value'>\"[html_encode(value)]\"</span>"
+		html += "[prefix]<span class='value'>\"[html_encode(value)]\"</span>"
 
 	else if (isicon(value))
 		#ifdef VARSICON
-		html += "[name] = /icon (<span class='value'>[value]</span>) [bicon(value)]"
+		html += "[prefix]/icon (<span class='value'>[value]</span>) [bicon(value)]"
 		#else
-		html += "[name] = /icon (<span class='value'>[value]</span>)"
+		html += "[prefix]/icon (<span class='value'>[value]</span>)"
 		#endif
 
 	else if(istype(value, /image))
 		#ifdef VARSICON
-		html += "<a href='?_src_=vars;Vars=\ref[value]'>[name] \ref[value]</a> = /image (<span class='value'>[value]</span>) [bicon(value)]"
+		html += "[prefix]<a href='?_src_=vars;Vars=\ref[value]'>/image (<span class='value'>[value]</span>) \ref[value] [bicon(value)]</a>"
 		#else
-		html += "<a href='?_src_=vars;Vars=\ref[value]'>[name] \ref[value]</a> = /image (<span class='value'>[value]</span>)"
+		html += "[prefix]<a href='?_src_=vars;Vars=\ref[value]'>/image (<span class='value'>[value]</span>) \ref[value]</a>"
 		#endif
 
 	else if (isfile(value))
-		html += "[name] = <span class='value'>'[value]'</span>"
+		html += "[prefix]<span class='value'>'[value]'</span>"
 
 	else if (istype(value, /datum))
 		var/datum/D = value
-		html += "<a href='?_src_=vars;Vars=\ref[value]'>[name] \ref[value]</a> = [D.type]"
+		html += "[prefix]<a href='?_src_=vars;Vars=\ref[value]'>[D.type] (<span class='value'>[D]</span>) \ref[value]</a>"
 
 	else if (istype(value, /client))
 		var/client/C = value
-		html += "<a href='?_src_=vars;Vars=\ref[value]'>[name] \ref[value]</a> = [C] [C.type]"
-//
-	else if (istype(value, /list))
+		html += "[prefix]<a href='?_src_=vars;Vars=\ref[value]'>[C.type] (<span class='value'>[C]</span>) \ref[value]</a>"
+
+	else if (islist(value))
 		var/list/L = value
-		html += "[name] = /list ([L.len])"
+		html += "[prefix]/list ([L.len])"
 
 		if (L.len > 0 && !(name == "underlays" || name == "overlays" || name == "vars" || L.len > 500))
 			// not sure if this is completely right...
 			html += "<ul>"
 			var/index = 1
 			for (var/entry in L)
-				if(istext(entry))
-					html += "<li>"+debug_variable(entry, L[entry], level + 1)
+				if(!(name == "contents" || name == "locs") && !isnum(entry) && !isnull(L[entry]))
```

Comment:
This is slightly wrong, but so is the similar line a few lines above, and I couldn't think of a simple way to fix either of them. Something did just occur to me while I was writing this comment, though. I'll try it tomorrow.

Reply:
I lied. I just did it.